packages/kit/src/runtime/client/router.js

```diff
@@ -190,12 +190,12 @@ export class Router {
 			// Removing the hash does a full page navigation in the browser, so make sure a hash is present
 			const [base, hash] = url.href.split('#');
 			if (hash !== undefined && base === location.href.split('#')[0]) {
-				// Call `pushState` to add url to history so going back works.
-				// Also make a delay, otherwise the browser default behaviour would not kick in
-				setTimeout(() => history.pushState({}, '', url.href));
+				event.preventDefault();
```

Comment:
This would disable the focus management done by the browser. Making https://github.com/sveltejs/kit/issues/3105 a problem again.

Reply:
Oh never mind. It's handled by hash setter and I tested with the wrong commit. But I notice the test for this is wrong. It actually won't fail.

Reply:
Yeah, I'll deal with it. thanks

Reply:
I mean this one https://github.com/sveltejs/kit/blob/2fefa5e9cf01510dd03c1a9cc0b62c9686621faa/packages/kit/test/apps/basics/test/test.js#L1732

There should be another focusable element between `<p id="p1">` and `<p id="p2">` here
https://github.com/sveltejs/kit/blob/4453c67a9526c3752d7eee7e53b7b1390c631e09/packages/kit/test/apps/basics/src/routes/routing/hashes/target.svelte#L7

I think it's somewhat unrelated to this PR. So you don't have to deal with it. I can create another PR for this later.